presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,345 @@
+                throw new IllegalArgumentException(format("Invalid dataFormat '%s' for column '%s'", dataFormat, columnName));
+            }
+            return FieldType.BYTE;
+        }
+
+        private static void checkFieldType(String columnName, Type columnType, FieldType fieldType)
+        {
+            if (columnType == BIGINT) {
+                checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT, FieldType.INT, FieldType.LONG);
+            }
+            else if (columnType == INTEGER) {
+                checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT, FieldType.INT);
+            }
+            else if (columnType == SMALLINT) {
+                checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT);
+            }
+            else if (columnType == TINYINT) {
+                checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE);
+            }
+            else if (columnType == BOOLEAN) {
+                checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT, FieldType.INT, FieldType.LONG);
+            }
+            else if (columnType == DOUBLE) {
+                checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.DOUBLE, FieldType.FLOAT);
+            }
+            else if (isVarcharType(columnType)) {
+                checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE);
+            }
+        }
+
+        private static void checkFieldTypeOneOf(FieldType declaredFieldType, String columnName, Type columnType, FieldType... allowedFieldTypes)
+        {
+            checkArgument(Arrays.asList(allowedFieldTypes).contains(declaredFieldType),
+                    format("Wrong dataformat '%s' specified for column '%s'; %s type implies use of %s",
+                            declaredFieldType.name(),
+                            columnName,
+                            columnType,
+                            Joiner.on("/").join(allowedFieldTypes)));
+        }
+
+        public String getName()
+        {
+            return name;
+        }
+
+        public Type getType()
+        {
+            return type;
+        }
+
+        public int getStart()
+        {
+            return start;
+        }
+
+        public int getEnd()
+        {
+            return end;
+        }
+
+        public FieldType getFieldType()
+        {
+            return fieldType;
+        }
+
+        public int getLength()
+        {
+            return end - start;
+        }
+    }
+
+    private static boolean isSupportedType(Type type)
+    {
+        return isVarcharType(type) || SUPPORTED_PRIMITIVE_TYPES.contains(type);
+    }
+
+    @Override
+    protected void appendNullValue()
+    {
+        for (int i = 0; i < columnMappings.get(currentColumnIndex).getLength(); i++) {
+            buffer.put((byte) 0);
```

Comment:
This is another missing part in the spec, right?  It doesn't say how nulls should be encoded.

Reply:
Correct. I could be interpreting this wrong but based on the code in the `RawColumnDecoder` it looks like the only case in which a column can be null is if `start` and `end` values for the mapping are equal. Guess I should remove `appendNullValue` for now and we can figure that out in a future commit.

Reply:
Ugh the append null method is used when no key value is specified

Reply:
> when no key value is specified

What do you mean by that. INSERT when only subset of columns are specified?

Can we just disallow writing NULLs into table with Raw encoders? Which usecases does it break?
It seems that with mapping we now support for write we would not be able to write a value which would be interpreted as NULL by decoder anyway.

Reply:
It wouldn't necessarily break anything, but you would have to insert values for every column when inserting. This includes the key column(s). It would definitely be best to disable writing null values until we can figure out a way to properly read/write them.